colossalai/shardformer/policies/gpt2.py

```diff
@@ -38,6 +38,13 @@ def preprocess(self):
             if vocab_size % world_size != 0:
                 new_vocab_size = vocab_size + world_size - vocab_size % world_size
                 self.model.resize_token_embeddings(new_vocab_size)
+        elif self.shard_config.pipeline_stage_manager is not None:
```

Comment:
I think this is not related to pipeline parallelism, right? As it actually affects kernels rather than p2p communication. Shall we make it more general?

Reply:
updated.